fw/ja5_conf.c

```diff
@@ -0,0 +1,243 @@
+/**
+ *		Tempesta FW
+ *
+ * Transport Layer Security (TLS) interfaces to Tempesta TLS.
+ *
+ * Copyright (C) 2024 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+#include <linux/list.h>
+#include <linux/hashtable.h>
+#include <linux/slab.h>
+
+#include "log.h"
+#include "ja5_conf.h"
+#include "hash.h"
+
+/* Define default size as multiple of TDB extent size */
+#define TLS_JA5_DEFAULT_STORAGE_SIZE ((1 << 21) * 25)
+#define TLS_JA5_HASHTABLE_BITS 10
+
+typedef struct {
+	struct hlist_node	hlist;
+	atomic64_t		refcnt;
+	/* TODO: make an unified macro for different types of ja5 hashes */
+	TlsJa5t			ja5_hash;
```

Comment:
`TlsJa5t` is not packed need to discuss on mitting

Reply:
I decided not to pack fingerprint in order to easily use it as a u64 key in hash tables.
Packing it seems low profitable because if you want to save 1 byte in it you always have to pack all structures that include it